### PR TITLE
pruning/add block mask feature

### PR DESCRIPTION
**Pull request by WeiweiZhang1**
## Type of Change

feature

API not changed

### CODE REVIEW

File: neural_compressor/compression/pruner/patterns.py [hunk 1]
```diff
@@ -19,11 +19,10 @@
 from .utils import torch
 from .utils import logger
 from collections import namedtuple
-
+import torch.nn.functional as F
```
Comment (wenhuach21) [inline neural_compressor/compression/pruner/patterns.py:22]: use lazy import, otherwise ut will fail

File: examples/pytorch/nlp/huggingface_models/question-answering/pruning/eager/run_qa_no_trainer_block.py
Comment (wenhuach21) [inline examples/pytorch/nlp/huggingface_models/question-answering/pruning/eager/run_qa_no_trainer_block.py:1]: please also help run one of the snip-moment examples to check whether there are any regressions
